src/Microsoft.AspNet.Mvc.Testing/ControllerTestHelper.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.FeatureModel;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Http.Interfaces;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.DependencyInjection.Fallback;
+
+namespace Microsoft.AspNet.Mvc.Testing
+{
+	public static class ControllerTestHelper
+	{
+		public static Controller Initialize(Controller controller, ControllerTestHelperContext context)
+		{
+			return Initialize(controller, new ActionDescriptor(), context);
+			// Question: Can we build (fully or partially) ActionDescriptor() from tyep information??
+		}
+
+		private static DefaultHttpContext CreateHttpContext(IServiceProvider serviceProvider)
+		{
+			var featureCollection = new FeatureCollection();
+			var featureContext = new FeatureContext();
+
+			featureCollection.Add(typeof(IHttpRequestLifetimeFeature), new FeatureContext());
+			featureCollection.Add(typeof(IHttpRequestFeature), new FeatureContext());
+
+			var httpContext = new DefaultHttpContext(featureCollection)
```

Comment:
The main reason of using this feature collection is providing users with a way to mock  Context.RequestAborted.IsCancellationRequested. I'll talk to @Tratcher if there is a better way to do this. In the meantime I'd like to keep this to unblock myself..